guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
+    ```bash
+    $ rails server
+    ```
+    This command starts your Rails server so you can view your application in
+    a web browser.
 
-Sprockets loads the files specified, processes them if
-necessary, concatenates them into one single file, and then compresses them
-(based on the value of `config.assets.css_compressor` or `config.assets.js_compressor`).
-Compression reduces file size, enabling the browser to download the files faster.
+6. Preview your application:
 
-### Controller Specific Assets
+    Open your web browser and navigate to `http://localhost:3000`. You should
+    see your Rails application with the included assets.
 
-When you generate a scaffold or a controller, Rails also generates a
-Cascading Style Sheet file for that controller. Additionally, when generating a scaffold, Rails generates the file `scaffolds.css`.
+### Manifest Files
 
-For example, if you generate a `ProjectsController`, Rails will also add a new
-file at `app/assets/stylesheets/projects.css`. By default, these files will be
-ready to use by your application immediately using the `link_directory` directive in the `manifest.js` file.
+In Propshaft, the `manifest.json` file is automatically generated during the
+asset precompilation process. This file maps original asset filenames to their
+fingerprinted versions, ensuring proper cache invalidation and efficient asset
+management. Located in the `public/assets` directory, the `manifest.json` file
+helps Rails resolve asset paths at runtime, allowing it to reference the correct
+fingerprinted files.
 
-You can also opt to include controller-specific stylesheets files
-only in their respective controllers using the following:
+The default `manifest.json` includes entries for main assets like
+`application.js` and `application.css` as well as other files, such as images.
+Here's an example of what the JSON might look like:
 
-```html+erb
-<%= stylesheet_link_tag params[:controller] %>
+```json
+{
+  "application.css": "application-6d58c9e6e3b5d4a7c9a8e3.css",
+  "application.js": "application-2d4b9f6c5a7c8e2b8d9e6.js",
+  "logo.png": "logo-f3e8c9b2a6e5d4c8.png",
+  "favicon.ico": "favicon-d6c8e5a9f3b2c7.ico"
+}
 ```
 
-When doing this, ensure you are not using the `require_tree` directive in your `application.css`, as that could result in your controller-specific assets being included more than once.
-
 ### Asset Organization
 
-Pipeline assets can be placed inside an application in one of three locations:
-`app/assets`, `lib/assets` or `vendor/assets`.
-
-* `app/assets` is for assets that are owned by the application, such as custom images or stylesheets.
+Propshaft organizes assets within the `app/assets` directory, which includes
+subdirectories like `images`, `javascripts`, and `stylesheets`. You can place
+your JavaScript, CSS, image files, and other assets into these directories, and
+Propshaft will manage them during the precompilation process.
```

Comment:
This feels like it's been repeated couple times already, but I guess it's okay since this section is about asset organization.

Reply:
Yup! Thanks for calling this out. This and a few other sections were repeated because of my structure, however, I’ve re-ordered the sections so that the flow feels much more natural, and it helped me catch and remove some duplications.